soc/arm/nordic_nrf/soc_nrf_common.h

```diff
@@ -66,14 +66,21 @@
  * @param prop_name human-readable string name for 'prop'
  */
 #define NRF_DT_CHECK_GPIO_CTLR_IS_SOC(node_id, prop, prop_name)		\
-	BUILD_ASSERT(!DT_NODE_HAS_PROP(node_id, prop) ||		\
-		     DT_NODE_HAS_COMPAT(DT_GPIO_CTLR(node_id, prop),	\
-					nordic_nrf_gpio),		\
-		     "Devicetree node " DT_NODE_PATH(node_id)		\
-		     " property " prop_name " must refer to a GPIO "	\
-		     " controller with compatible nordic,nrf-gpio; "	\
-		     "got " DT_NODE_PATH(DT_GPIO_CTLR(node_id, prop))	\
-		     ", which does not have this compatible")
+	COND_CODE_1(DT_NODE_HAS_PROP(node_id, prop),			\
+		   (BUILD_ASSERT(DT_NODE_HAS_COMPAT(			\
+					 DT_GPIO_CTLR(node_id, prop),	\
+					 nordic_nrf_gpio),		\
+				 "Devicetree node "			\
+				 DT_NODE_PATH(node_id)			\
+				 " property " prop_name			\
+				 " must refer to a GPIO controller "	\
+				 "with compatible nordic,nrf-gpio; "	\
+				 "got "					\
+				 DT_NODE_PATH(DT_GPIO_CTLR(node_id,	\
+							   prop))	\
+				 ", which does not have this "		\
+				 "compatible")),			\
+		    ((void)0))	/* allow a trailing ; either way */
```

Comment:
I don't understand the need of this `(void)0`. Are users supposed to place this macro inside a function? It will give a compilation error if [used as in #31965](https://github.com/zephyrproject-rtos/zephyr/pull/31965/files#diff-c5892bc118bf2fdb8dad83050134cf9807b1725f50c224784a8fa275edb2b1ffR331) when the checked property is not present.

Reply:
Yes, that's a good point.  This macro has to be used at file scope or statement scope, so it needs a different "ignore-me" case, like

    extern char _this_does_not_exist

I figured it wasn't necessary to ask for a test, but now I think that thunk was bad.

Reply:
> it needs a different "ignore-me" case

Can't it be just empty?

Reply:
Yes.  I thought that having an empty declaration at file scope would be syntactically wrong, but it seems to be OK.

Reply:
That's weird, I build tested this with and without the relevant properties set yesterday. Let me look further.

Reply:
> Can't it be just empty?

Isn't an empty statement with an extraneous semicolon a linter issue in some configurations? That's why I ended up going with a BUILD_ASSERT() in both cases.